### PR TITLE
feat: new plugin for gitea

### LINKED ISSUE #2992: [Feature][Plugin] New plugin for Gitea
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-devlake/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

[Gitea](https://gitea.io/) is an open-source software package for hosting software development version control using Git as well as other collaborative features like bug tracking, code review, kanban boards, tickets, and wikis.

### Use case

Gitea plugin

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by tk103331**
# Summary
[Gitea](https://gitea.io/) is an open-source software package for hosting software development version control using Git as well as other collaborative features like bug tracking, code review, kanban boards, tickets, and wikis.

### Does this close any open issues?
Closes [2992](https://github.com/apache/incubator-devlake/issues/2992)

### Screenshots
Include any relevant screenshots here.

### Other Information
Any other information that is important to this PR.

### CODE REVIEW

**Comment by warren830** [comment]
Amazing work, I will review it later, thanks!

**Comment by keon94** [comment]
Let's hold this off until I merge #2907. We will be enforcing errors.Error to be the return type of all functions in Devlake soon.

File: plugins/gitea/tasks/repo_collector.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		PageSize:           50,
+		UrlTemplate:        "repos/{{ .Params.Owner }}/{{ .Params.Repo }}",
+		Query: func(reqData *helper.RequestData) (url.Values, error) {
+			query := url.Values{}
+			query.Set("state", "all")
+			query.Set("page", fmt.Sprintf("%v", reqData.Pager.Page))
+			query.Set("limit", fmt.Sprintf("%v", reqData.Pager.Size))
+			return query, nil
+		},
+		ResponseParser: func(res *http.Response) ([]json.RawMessage, error) {
```
Comment (warren830) [inline plugins/gitea/tasks/repo_collector.go:57]: We should add a func: GetTotalPages

**Comment by RTZE23** [comment]
@warren830 I would love to use this plugin. Is it possible to merge the feature branch into the main branch? It looks like nothing has happened here since september.

**Pull request merged**